pkg/runtime/serializer/recognizer/recognizer.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package recognizer
```

Comment:
It's a content sniffer - it would look for opening brace, magic number for
protobuf, or yaml begin doc.  Sniffer felt too weird as package name.  Not
married to any particular name.


On Dec 1, 2015, at 7:37 PM, Daniel Smith <notifications@github.com> wrote:

In pkg/runtime/serializer/recognizer/recognizer.go
<https://github.com/kubernetes/kubernetes/pull/17922#discussion_r46362645>:

> +Copyright 2014 The Kubernetes Authors All rights reserved.
> +
> +Licensed under the Apache License, Version 2.0 (the "License");
> +you may not use this file except in compliance with the License.
> +You may obtain a copy of the License at
> +
> +    http://www.apache.org/licenses/LICENSE-2.0
> +
> +Unless required by applicable law or agreed to in writing, software
> +distributed under the License is distributed on an "AS IS" BASIS,
> +WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
> +See the License for the specific language governing permissions and
> +limitations under the License.
> +*/
> +
> +package recognizer

I don't understand why this package and file are called "recognizer"?

—
Reply to this email directly or view it on GitHub
<https://github.com/kubernetes/kubernetes/pull/17922/files#r46362645>.